core/controllers/profile.py

```diff
@@ -101,27 +102,76 @@ def get(self, username):
         self.render_json(self.values)
 
 
-class BulkEmailWebhookEndpoint(base.BaseHandler):
+class BulkEmailWebhookEndpoint(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
     """The endpoint for the webhook that is triggered when a user
     subscribes/unsubscribes to the bulk email service provider externally.
     """
 
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'secret': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'POST': {
+            'data[list_id]': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'data[email]': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_regex_matched',
+                        'regex_pattern': constants.EMAIL_REGEX
+                    }]
+                },
+                'default_value': None
```

Comment:
Why can this be `None`?

Reply:
Because if list_id is not equal to `feconf.MAILCHIMP_AUDIENCE_ID` then we no need to pass email_id and type.

Reply:
default None removed!